template/before.sh.erb

```diff
@@ -4,3 +4,129 @@
 # Find available port to run server on
 port=8080
 
+# Define Galaxy Variables
+export GALAXY_PATH="<%= BatchConnect::App.from_token(session.token).root.realpath.join("galaxy").to_s %>"
+export GALAXY_CONFIG_FILE="${PWD}/galaxy.yml"
+MOUNT="/node/${HOSTNAME}/8080=galaxy.webapps.galaxy.buildapp:uwsgi_app()"
+DATAROOT=${HOME}/.galaxy
+EMAIL="${USER}@osc.edu"
+export JOB_CONFIG_FILE_PATH="${PWD}/job_conf.xml"
+export JOB_RESOURCE_PARAMS_CONF_FILE_PATH="${PWD}/job_resource_params_conf.xml"
+export WORKFLOW_RESOURCE_PARAMS_CONF_FILE_PATH="${PWD}/workflow_resource_params_conf.xml"
+
+echo "GALAXY location: '${GALAXY_PATH}'"
+echo "Data location: '${DATAROOT}'"
+
+# Load the required environment
+<%-
+  if context.node_type=="hugemem"
+      ppn = 48
+  elsif context.num_cores.blank?
+      ppn = 28
+  else
+      ppn = context.num_cores.to_i
+  end
+-%>
+
+# Generate Galaxy job configuration file
+(
+umask 077
+cat > "${JOB_CONFIG_FILE_PATH}" << EOL
+<?xml version="1.0"?>
+<!-- A sample job config that explicitly configures job running the way it is configured by default (if there is no explicit config). -->
+<job_conf>
+    <resources>
+        <group id="basic">cores,time</group>
+    </resources>
+    <tools>
+        <tool id="bed2gff1" destination="dynamic_cores_time" resources="basic"/>
+    </tools>
+    <plugins>
+        <plugin id="local" type="runner" load="galaxy.jobs.runners.local:LocalJobRunner"/>
+        <plugin id="pbs" type="runner" load="galaxy.jobs.runners.pbs:PBSJobRunner"/>
+        <plugin id="dynamic" type="runner">
+            <param id="rules_module">galaxy.jobs.rules</param>
+        </plugin>
+    </plugins>
+    <destinations default="<%= context.job_runner %>">
+        <destination id="dynamic_cores_time" runner="dynamic">
+            <param id="type">python</param>
+            <param id="function">dynamic_cores_time</param>
+        </destination>
+        <destination id="pbs" runner="pbs">
+            <param id="Resource_List">walltime=5:00:00,nodes=1:ppn=<%= ppn %></param>
```

Comment:
Maybe we should default the walltime to 1 hour and ppn=1, since this is the job runner. We should assume tools are simple and give the option for users to request more. So my understanding with the parameters below is that we are basically saying you can submit a job to an Owens compute node, and request 1 core or all 28 cores and also change the walltime.